spec/support/shared_examples/provider_auth_shared_examples.rb

```diff
@@ -0,0 +1,6 @@
+RSpec.shared_examples 'a provider not authenticated' do
+  it 'should redirect the user to the login page' do
+    perform_request
+    expect(response).to redirect_to(new_provider_session_path)
+  end
+end
```

Comment:
If you refactor this so as to pass in the request object:
```
RSpec.shared_examples 'a provider not authenticated' do | current_request |
  it 'should redirect the user to the login page' do
    current_request
    expect(response).to redirect_to(new_provider_session_path)
  end
end
```
Then in a spec the code would be:
```
it_behaves_like 'a provider not authenticated', perform_request
```
That way the request object doesn't always have to be labelled `perform_request`.